code/game/objects/items/ashtray.dm

```diff
@@ -1,17 +1,31 @@
-/obj/item/ashtray
+/obj/item/storage/ashtray
 	icon = 'icons/ashtray.dmi'
-	var/max_butts = 0
+	w_class = WEIGHT_CLASS_SMALL
+	can_hold = list(/obj/item/cigbutt, /obj/item/clothing/mask/cigarette)
+	allow_quick_empty = TRUE
+	allow_quick_gather = FALSE
+
+	// No generic container insertion messages/sounds.
+	silent = TRUE
+	use_sound = null
+
 	var/icon_half = ""
 	var/icon_full = ""
 
-/obj/item/ashtray/Initialize(mapload)
+/obj/item/storage/ashtray/Initialize(mapload)
 	. = ..()
 	pixel_y = rand(-5, 5)
 	pixel_x = rand(-6, 6)
 
-/obj/item/ashtray/attackby(obj/item/I, mob/user, params)
+/obj/item/storage/ashtray/attack_self(mob/user)
+	// We want allow_quick_empty to enable disposal dumping
+	// but don't want to accidentally dump ashtray contents
+	// on the floor.
+	return
+
+/obj/item/storage/ashtray/attackby(obj/item/I, mob/user, params)
```

Comment:
I am not sure this is needed anymore. surely it being a storage item, this is handled already?